test/system/src/test/java/io/pravega/test/system/KvsWithFailOverTest.java

```diff
@@ -0,0 +1,140 @@
+    public Timeout globalTimeout = Timeout.seconds(25 * 60);
+    private StreamManager streamManager;
+    private ConnectionFactory connectionFactory;
+    private ConnectionPool connectionPool;
+    private KeyValueTableFactory keyValueTableFactory;
+
+    @Environment
+    public static void initialize() throws MarathonException, ExecutionException {
+        URI zkUri = startZookeeperInstance();
+        startBookkeeperInstances(zkUri);
+    }
+
+    @Before
+    public void setup() throws ExecutionException {
+        // Get zk details to verify if controller, SSS are running
+        Service zkService = Utils.createZookeeperService();
+        List<URI> zkUris = zkService.getServiceDetails();
+        log.debug("Zookeeper service details: {}", zkUris);
+        //get the zk ip details and pass it to  host, controller
+        URI zkUri = zkUris.get(0);
+        // Verify controller is running.
+        controllerInstance = Utils.createPravegaControllerService(zkUri);
+        assertTrue(controllerInstance.isRunning());
+        List<URI> conURIs = controllerInstance.getServiceDetails();
+        log.info("Pravega Controller service instance details: {}", conURIs);
+
+        // Fetch all the RPC endpoints and construct the client URIs.
+        final List<String> uris = conURIs.stream().filter(ISGRPC).map(URI::getAuthority).collect(Collectors.toList());
+        controllerURIDirect = URI.create("tcp://" + String.join(",", uris));
+        log.info("Controller Service direct URI: {}", controllerURIDirect);
+        // Verify segment store is running.
+        segmentStoreInstance = Utils.createPravegaSegmentStoreService(zkUri, controllerURIDirect);
+        assertTrue(segmentStoreInstance.isRunning());
+        log.info("Pravega Segmentstore service instance details: {}", segmentStoreInstance.getServiceDetails());
+        executorService = ExecutorServiceHelpers.newScheduledThreadPool(NUM_KVPWRITE + NUM_KVPREAD + 1, "KvsFailOverTest-main");
+        controllerExecutorService = ExecutorServiceHelpers.newScheduledThreadPool(2, "KvsFailOverTest-controller");
+        final ClientConfig clientConfig = Utils.buildClientConfig(controllerURIDirect);
+        log.info("BEFORE 1");
+        // Making controller and segmentstore instance count to 1
+        //log.info("controller size {}", (controllerInstance.getServiceDetails().size()) / 2);
+        //if (((controllerInstance.getServiceDetails().size()) / 2) > 1)
+        Futures.getAndHandleExceptions(controllerInstance.scaleService(1), ExecutionException::new);
+        log.info("BEFORE 2");
+        //log.info("Segmentstore size {}", segmentStoreInstance.getServiceDetails().size());
+        //if (segmentStoreInstance.getServiceDetails().size() > 1)
+        Futures.getAndHandleExceptions(segmentStoreInstance.scaleService(1), ExecutionException::new);
+        log.info("BEFORE 3");
```

Comment:
Remove all of these BEFORE logs. They add no value.

Reply:
Resolved